src/kernel.rs

```diff
@@ -2,13 +2,19 @@
 
 /// An borrowed 2D kernel, used to filter images via convolution.
 #[derive(Debug, Copy, Clone)]
-pub struct Kernel<'a, K> {
+pub struct Kernel<'a, K>
+where
+    K: Copy,
+{
```

Comment:
Having bounds in struct definitions is bad practice, bounds should appear where they are used.